### PR TITLE
fixing loading state in menubar. fixing key problems in renderer also

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers 
This fixes some errant cpu load when you startup and don't open the widget
also fixed a warning due to children w/o keys

### CODE REVIEW

File: shared/menubar/index.js [hunk 1]
```diff
@@ -28,12 +28,17 @@ export type MenubarProps = {
...
+const REQUEST_DELAY = 5000
+
 class Menubar extends Component {
   props: MenubarProps;
+  _lastRequest: number;
 
   constructor (props) {
     super(props)
 
+    this._lastRequest = 0
```
Comment (chrisnojima) [inline shared/menubar/index.js]: we keep track of the last request we made so we don't hammer the rpc

File: shared/menubar/index.js [hunk 2]
```diff
@@ -81,6 +94,21 @@ class Menubar extends Component {
     this.checkForFolders()
   }
 
+  shouldComponentUpdate (nextProps, nextState) {
+    if (this.props.username !== nextProps.username ||
+        this.props.loggedIn !== nextProps.loggedIn) {
+      this._lastRequest = 0 // reset delay since user/loggedin changed
```
Comment (chrisnojima) [inline shared/menubar/index.js]: if our user or logged in state changed, lets get the folders

File: shared/menubar/index.js [hunk 3]
```diff
@@ -81,6 +94,21 @@ class Menubar extends Component {
...
+      this.checkForFolders()
+      return true
+    }
+
+    if (this.props.folders !== nextProps.folders) {
+      return true
```
Comment (chrisnojima) [inline shared/menubar/index.js]: if the folders changed, lets render

File: shared/menubar/index.render.desktop.js [hunk 4]
```diff
@@ -116,16 +116,16 @@ const Row = props => {
...
-      <Text type='Body' link small key={props.key} style={{marginTop: 4, overflowWrap: 'break-word', flex: 1, ...props.textStyle}}>{props.text}</Text>
+      <Text type='Body' link small style={{marginTop: 4, overflowWrap: 'break-word', flex: 1, ...props.textStyle}}>{props.text}</Text>
       {props.children}
     </div>
   )
 }
 
 const FolderRow = props => {
-  const divider = <span>,<wbr/></span> // word break on the commas
+  const divider = idx => <span key={'div' + idx}>,<wbr/></span> // word break on the commas
```
Comment (chrisnojima) [inline shared/menubar/index.render.desktop.js]: we have to use the dividerFn to get keyd dividers ,else all the dividers will get thrashed

**Comment by MarcoPolo** [comment]
This menubar keeps getting hackier :P Sagas would probably help clean this up.

Probably not worth spending the time to fix this just yet though.

:+1:

**Pull request merged**